src/schema-association-service-1espt.ts

```diff
@@ -0,0 +1,84 @@
+                const config = vscode.workspace.getConfiguration('azure-pipelines')
+                config.update('1ESPipelineTemplatesSchemaFile', false, vscode.ConfigurationTarget.Workspace); // disable the 1ESPT schema configuration
+                vscode.window.showInformationMessage(Messages.disabled1ESPTSchemaAsADOOrgNotContains1ESPT)
+                return undefined; // 1ESPT repo not found
+            }
+            const repository = repositories.find(repo => repo.name === "1ESPipelineTemplates");
+
+            // Using getItem from GitApi: getItem(repositoryId: string, path: string, project?: string, scopePath?: string, recursionLevel?: GitInterfaces.VersionControlRecursionType, includeContentMetadata?: boolean, latestProcessedChange?: boolean, download?: boolean, versionDescriptor?: GitInterfaces.GitVersionDescriptor, includeContent?: boolean, resolveLfs?: boolean, sanitize?: boolean): Promise<GitInterfaces.GitItem>;
+            var schemaFile = await gitApi.getItem(repository.id, "schema/1espt-base-schema.json", "1ESPipelineTemplates", undefined, undefined, true, true, true, undefined, true, true)
+
+            const schemaContent = schemaFile.content;
+            const schemaUri = Utils.joinPath(context.globalStorageUri, '1ESPTSchema', `${organizationName}-1espt-schema.json`);
+            await vscode.workspace.fs.writeFile(schemaUri, Buffer.from(schemaContent));
+            return schemaUri;
+        }
+        else
+        // if user is signed in with account other than microsoft, then disable 1ESPT schema and delete the 1ESPT schema file
+        {
+            const config = vscode.workspace.getConfiguration('azure-pipelines', workspaceFolder)
+            config.update('1ESPipelineTemplatesSchemaFile', false, vscode.ConfigurationTarget.Workspace);
+            await vscode.workspace.fs.delete(Utils.joinPath(context.globalStorageUri, '1ESPTSchema'), { recursive: true })
+        }
+    }
+    catch (error) {
+        logger.log(`Error : ${error} while fetching 1ESPT schema for org: ${organizationName}  : `, 'SchemaDetection');
+    }
+    return undefined;
+}
+
+export async function getCached1ESPTSchemaInformation(context: vscode.ExtensionContext, organizationName: string, session: AzureSession, lastUpdated1ESPTSchema: Map<string, Date>, seen1ESPTOrganizations: Set<string>): Promise<URI> {
+    if (seen1ESPTOrganizations.has(organizationName)) {
+        const schemaUri1ESPT = Utils.joinPath(context.globalStorageUri, '1ESPTSchema', `${organizationName}-1espt-schema.json`);
+
+        try {
+            if (session.userId.endsWith("@microsoft.com") || session.userId.endsWith(".microsoft.com")) {
+                if ((new Date().getTime() - lastUpdated1ESPTSchema.get(organizationName).getTime()) < milliseconds24hours) {
+                    var schemaFileExists = await vscode.workspace.fs.stat(schemaUri1ESPT);
+                    if (schemaFileExists) {
+                        // Fetch cached 1ESPT schema if:
```

Comment:
Super tiny nit: this is a really helpful comment, in fact helpful enough that I think it should become the function's JSDoc documentation instead of being inlined here :).

Reply:
makes sense. Done